src/scripts/import-needs-assessment-data/add-surveys.ts

```diff
@@ -0,0 +1,288 @@
+    // if (key !== UploadWorkflowStatus.SUCCESS && key !== UploadWorkflowStatus.ALREADY_EXISTS) {
+    //   resultsMap[key].forEach((result) => {
+    //     console.log(result)
+    //     console.log("\n")
+    //   })
+    // }
+  });
+
+  console.log("Adding items completed!");
+
+  const validSurveys: Survey[] = [
+    ...resultsMap[UploadWorkflowStatus.SUCCESS],
+    ...resultsMap[UploadWorkflowStatus.ALREADY_EXISTS],
+  ].reduce((surveys: Survey[], workflow: SurveyUploadWorkflow) => {
+    return [...surveys, workflow.data];
+  }, [] as Survey[]);
+
+  return validSurveys;
+}
+
+const isFulfilled = <T>(
+  value: PromiseSettledResult<T>,
+): value is PromiseFulfilledResult<T> => {
+  return value.status === "fulfilled";
+};
+
+const _isRejected = <T>(
+  value: PromiseSettledResult<T>,
+): value is PromiseRejectedResult => {
+  return value.status === "rejected";
+};
+
+/*  Consolidate Surveys
+ * ------------------------------------------------------- */
+function consolidateSurveys(data: NeedAssessment[]): string[] {
+  const uniqueSurveys = new Set<string>();
+
+  data.forEach((item) => {
+    const surveyString = `${item.survey.year}-${item.survey.quarter}`;
+    uniqueSurveys.add(surveyString);
+  });
+
+  return Array.from(uniqueSurveys);
+}
+
+/*  Parse Surveys
+ * ------------------------------------------------------- */
+function parseSurvey({
+  data,
+  orig,
+  status,
+  logs,
+}: SurveyUploadWorkflow): SurveyUploadWorkflow {
+  logs = [...logs, `Log: parsing survey "${orig}"`];
+
+  if (typeof orig === "string") {
+    //if orig is a string, validate it
+    const [year, quarter] = orig.split("-");
+    if (!year || !quarter) {
+      throw {
+        data,
+        orig,
+        status: UploadWorkflowStatus.ORIGINAL_DATA_INVALID,
+        logs: [
+          ...logs,
+          `Error: Invalid survey input: "${orig}". Expected a non-null string in YYYY-QQ format.`,
+        ],
+      };
+    }
+  } else if (typeof orig === "object" && orig !== null) {
+    // if orig is an object, validate it
+    if (!orig.year || !orig.quarter) {
+      throw {
+        data,
+        orig,
+        status: UploadWorkflowStatus.ORIGINAL_DATA_INVALID,
+        logs: [
+          ...logs,
+          `Error: Invalid survey input: "${JSON.stringify(orig)}". Expected a non-null object with year and quarter.`,
+        ],
+      };
+    }
+  } else {
+    throw {
+      data,
+      orig,
+      status: UploadWorkflowStatus.ORIGINAL_DATA_INVALID,
+      logs: [
+        ...logs,
+        `Error: Invalid survey input: "${orig}". Expected a non-null value.`,
+      ],
+    };
+  }
+
+  const newData = {
+    ...data,
+    survey: orig,
+  };
+
+  return {
+    data: newData,
+    orig,
+    status,
+    logs,
+  };
+}
+
+/*  Get Surveys
+ * ------------------------------------------------------- */
+async function getSurvey({
+  data,
+  orig = `${data.year}-${data.quarter}`,
```

Comment:
**question (blocking):** Is there a specific reason for doing this?  The point of passing the `orig` data through is to have a copy of it from before it was processed, to compare to the processed `data` for debugging.  It also looks like `orig` is in the YYYY-QQ format since it comes from `consolidateSurveys`

Reply:
I was having issues with types when I was originally testing the functions on their own, so this likely came into play then when working with the data and ended up remaining instead of being removed. My bad. 🤦🏻‍♀️